component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthenticator.java

```diff
@@ -199,4 +212,119 @@ private List<String> populateMandatoryClaims() {
         mandatoryClaims.add(JWT_ID_CLAIM);
         return mandatoryClaims;
     }
+
+    /**
+     * Validate whether the request signing algorithm is registered for the application.
+     *
+     * @param request                 Http servlet request.
+     * @param bodyParameters          Map of request body params.
+     * @param oAuthClientAuthnContext OAuthClientAuthnContext.
+     * @return whether the request signing algorithm is registered for the application.
+     * @throws OAuthClientAuthnException
+     */
+    public boolean isRegisteredSignatureAlgorithm(HttpServletRequest request,
+                                                  Map<String, List> bodyParameters,
+                                                  OAuthClientAuthnContext oAuthClientAuthnContext)
+            throws OAuthClientAuthnException{
+
+        String signedObject = request.getParameter(OAUTH_JWT_ASSERTION);
+        if (isNotEmpty(signedObject)) {
+            //   Obtain the signing algorithm used to sign the JWT in the request
+            String requestSigningAlgorithm = getRequestSigningAlgorithm(signedObject);
+            //   Obtain the signing algorithm registered for the application
+            String registeredSigningAlgorithm = getRegisteredSigningAlgorithm(getClientId(request, bodyParameters,
+                    oAuthClientAuthnContext));
+            if (registeredSigningAlgorithm.equals(Constants.NOT_APPLICABLE)) {
+                return false;
+            }
+            //   Mandating PS256 and ES256 as the JWT signing algorithms
+            if (!(Constants.ALG_ES256.equals(requestSigningAlgorithm) ||
+                    Constants.ALG_PS256.equals(requestSigningAlgorithm))) {
+                if (log.isDebugEnabled()) {
```

Comment:
We better add an error log

Reply:
Fixed in 11fcdc02de410f4830875ab9f3869f53358e0027